### ISSUE TITLE
reversing arm64 files cause process of Ninja crash

### ISSUE THREAD

**Issue by bb33bb** [Vector35/binaryninja-api#6043]
**Version and Platform (required):**
- Binary Ninja Version: 4.2.6298-dev Personal (3fb67302)
- OS: manjaro 
- OS Version: Linux version 6.10.13-3-MANJARO (builduser@fv-az1246-770) (gcc (GCC) 14.2.1 20240910, GNU ld (GNU Binutils) 2.43.0) #1 SMP PREEMPT_DYNAMIC Tue Oct  8 03:24:49 UTC 2024
- CPU Architecture:x64

**Bug Description:**
crash

```
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Downloading source file /usr/src/debug/glibc/glibc/io/../sysdeps/unix/sysv/linux/ppoll.c
0x000072db5cf1abb0 in __GI_ppoll (fds=0x58ea344b3140, nfds=1,                   
    timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
42	  return SYSCALL_CANCEL (ppoll_time64, fds, nfds, timeout, sigmask,
(gdb) c
Continuing.

Thread 27 "W INT Flow Grap" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x72d9e52006c0 (LWP 117020)]
0x000072db5fc300f1 in BNHighLevelILTokenEmitterHasCollapsableRegions ()
   from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
(gdb) bt
#0  0x000072db5fc300f1 in BNHighLevelILTokenEmitterHasCollapsableRegions () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#1  0x000072db3c311ce1 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
#2  0x000072db3c290629 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
#3  0x000072db3c2963d9 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
#4  0x000072db3c3346f5 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
#5  0x000072db5fd49089 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#6  0x000072db5fa07ad0 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#7  0x000072db5fa0a474 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#8  0x000072db5fd49679 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#9  0x000072db5fbcac22 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#10 0x000072db5fa7ae40 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#11 0x000072db602ab48f in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
#12 0x000072db5d0e1c34 in std::execute_native_thread_routine (__p=0x58ea397f03e0) at /usr/src/debug/gcc/gcc/libstdc++-v3/src/c++11/thread.cc:104
#13 0x000072db5cea339d in start_thread (arg=<optimized out>) at pthread_create.c:447
#14 0x000072db5cf2849c in __GI___clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:78
(gdb) bt full
#0  0x000072db5fc300f1 in BNHighLevelILTokenEmitterHasCollapsableRegions () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#1  0x000072db3c311ce1 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
No symbol table info available.
#2  0x000072db3c290629 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
No symbol table info available.
#3  0x000072db3c2963d9 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
No symbol table info available.
#4  0x000072db3c3346f5 in ?? () from /mnt/4t/opt/binaryninja/plugins/liblang_pseudoc.so
No symbol table info available.
#5  0x000072db5fd49089 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#6  0x000072db5fa07ad0 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#7  0x000072db5fa0a474 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#8  0x000072db5fd49679 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#9  0x000072db5fbcac22 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#10 0x000072db5fa7ae40 in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#11 0x000072db602ab48f in ?? () from /mnt/4t/opt/binaryninja/libbinaryninjacore.so.1
No symbol table info available.
#12 0x000072db5d0e1c34 in std::execute_native_thread_routine (__p=0x58ea397f03e0) at /usr/src/debug/gcc/gcc/libstdc++-v3/src/c++11/thread.cc:104
--Type <RET> for more, q to quit, c to continue without paging--c
        __t = <optimized out>
#13 0x000072db5cea339d in start_thread (arg=<optimized out>) at pthread_create.c:447
        ret = <optimized out>
        pd = <optimized out>
        out = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {126280177551040, 7220515960802288412, 126280177551040, -3592, 2, 140724196195968, 7220515960789705500, 7219585664422028060}, 
              mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#14 0x000072db5cf2849c in __GI___clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:78
No locals.
```

regs
```
(gdb) i r
rax            0x0                 0
rbx            0x72d58b877f80      126261494513536
rcx            0x0                 0
rdx            0x72d9e51fe900      126280177543424
rsi            0x84                132
rdi            0x72d58b877f90      126261494513552
rbp            0x72d9e51fe900      0x72d9e51fe900
rsp            0x72d9e51fe550      0x72d9e51fe550
r8             0x5                 5
r9             0x72d9e51fe500      126280177542400
r10            0x72d9e51fe550      126280177542480
r11            0x83                131
r12            0x72d9780ee560      126278347711840
r13            0x72d9e51fe670      126280177542768
r14            0x72d9e51fe670      126280177542768
r15            0x72d9e51fe900      126280177543424
rip            0x72db5fc300f1      0x72db5fc300f1 <BNHighLevelILTokenEmitterHasCollapsableRegions+17>
eflags         0x10246             [ PF ZF IF RF ]
cs             0x33                51
ss             0x2b                43
ds             0x0                 0
es             0x0                 0
fs             0x0                 0
gs             0x0                 0
fs_base        0x72d9e52006c0      126280177551040
--Type <RET> for more, q to quit, c to continue without paging--c
gs_base        0x0                 0
```

**Comment by plafosse** [comment]
thanks for the report I have a fix incoming.

**Comment by bb33bb** [comment]
> thanks for the report I have a fix incoming.

appreciate !!
and waiting for the update, because i can not go on research without this udpate.
allways crash.

**Comment by bb33bb** [comment]
here is the trigger method
01-use mouse point to a function
02-then the preview will appear and then , process will crash.
![image](https://github.com/user-attachments/assets/332d3fbe-2e49-4108-ba7d-53c78dbc190e)

**Comment by plafosse** [comment]
THanks! Should be fixed in 4.2.6300
